bcgov/HOUS-permit-portal
# Feature/bphh 1236/external api admin screens

## Description
Feature/bphh 1236/external api admin screens
- Adds external api key index screen with a table of api keys for jurisdictions. Viewable as super admin and admin
- Adds ability to enable or disable api keys for a whole jurisdiction as a super admin
- Add's ability to create, update, expire and revoke api key's as super admin and review manager is api keys are enabled for the jurisdiction


## What type of PR is this? (check all applicable)

- [x ] 🍕 Feature

## Related Tickets & Documents
- https://hous-bssb.atlassian.net/browse/BPHH-1236
- https://hous-bssb.atlassian.net/browse/BPHH-1237


## Steps to QA
- First enable api keys for a jurisdiction as a Super admin
    - Navigate to jurisdictions page
    -  Select any jurisdiction and click manage -> External Api keys
    - Should navigate to index screen
    - Enable api key's using the switch top right of screen
   
![Screenshot 2024-04-25 at 5 12 25 PM](https://github.com/bcgov/HOUS-permit-portal/assets/32022335/c5027128-f361-4495-a794-8c2ea865ce66)
- To perform CRUD operations for external api key when api keys are enabled for the jurisdiction:
    - Navigate to external api key index page 
        - As a super admin - Navigate to Jurisdictions -> Click Manage -> External API keys
        - As a review manager - Navigate to Configuration Management -> Click edit on "Manage Configuration Management"
    - To Create click "Create new API key " button which should open up a modal
        - Add required details then click "Create"
        - On success should navigate to Manage modal
     - To Manage/Edit External API key, click "Manage" on any of the table keys
         - Should open up the Manage modal
         - Should be able to change everything except token, and revoked at
         - Should be able to revoke if already not revoked.  Should show a confirmation modal
         - And save
 
![Screenshot 2024-04-25 at 5 22 20 PM](https://github.com/bcgov/HOUS-permit-portal/assets/32022335/f16359cf-27f3-4f4a-9028-67e4384b4540)

![Screenshot 2024-04-25 at 5 22 26 PM](https://github.com/bcgov/HOUS-permit-portal/assets/32022335/83fc00a2-b9b1-4ef2-9690-240a2762f617)

https://github.com/bcgov/HOUS-permit-portal/assets/32022335/a7480500-b3ae-4390-a26d-f8e3a4d19c40

- to disable api keys for a jurisdiction as a Super admin
    - Navigate to jurisdictions page
    - Select any jurisdiction and click manage -> External Api keys
    - Should navigate to index screen
    - Disable api key's using the switch top right of screen
    - The Create and Manage buttons should become disabled
    - Should be the same for review manager once disabled, but should also show a warning banner
 
![Screenshot 2024-04-25 at 5 12 00 PM](https://github.com/bcgov/HOUS-permit-portal/assets/32022335/94ede139-99f6-4b7d-b33e-27451af7654e)

https://github.com/bcgov/HOUS-permit-portal/assets/32022335/af43b674-aaac-4cbb-9ad2-926ebcd9767a
